server/game/gameSystems/DeckSearchSystem.ts

```diff
@@ -0,0 +1,284 @@
+                    this.selectCard(event, additionalProperties, cards, newSelectedCards);
+                } else {
+                    this.handleDone(properties, context, event, newSelectedCards, cards);
+                }
+            }
+        });
+    }
+
+    private handleDone(properties: IDeckSearchProperties, context: AbilityContext, event: DeckSearchEvent, selectedCards: Set<Card>, allCards: Card[]): void {
+        event.selectedCards = Array.from(selectedCards);
+        context.selects['deckSearch'] = Array.from(selectedCards);
+        if (properties.selectedCardsHandler === null) {
+            this.defaultDoneHandle(properties, context, event, selectedCards);
+        } else {
+            properties.selectedCardsHandler(context, event, Array.from(selectedCards));
+        }
+
+        if (typeof properties.remainingCardsHandler === 'function') {
+            const cardsToMove = allCards.filter((card) => !selectedCards.has(card));
+            properties.remainingCardsHandler(context, event, cardsToMove);
+        } else {
+            this.defaultRemainingCardsHandler(properties, context, event, selectedCards, allCards);
+        }
+    }
+
+    private defaultRemainingCardsHandler(properties: IDeckSearchProperties, context: AbilityContext, event: DeckSearchEvent, selectedCards: Set<Card>, allCards: Card[]) {
+        if (this.shouldShuffle(properties.shuffle, context)) {
+            return event.player.shuffleDeck();
+        }
+
+        if (properties.placeOnBottomInRandomOrder) {
+            const cardsToMove = allCards.filter((card) => !selectedCards.has(card));
+            if (cardsToMove.length > 0) {
+                shuffleArray(cardsToMove);
+                for (const card of cardsToMove) {
+                    event.player.moveCard(card, Location.Deck, { bottom: true });
+                }
+                context.game.addMessage(
+                    '{0} puts {1} card{2} on the bottom of their conflict deck',
+                    event.player,
+                    cardsToMove.length,
+                    cardsToMove.length > 1 ? 's' : ''
+                );
+            }
+        }
+    }
+
+    private defaultDoneHandle(properties: IDeckSearchProperties, context: AbilityContext, event: DeckSearchEvent, selectedCards: Set<Card>): void {
+        this.handleDoneMessage(properties, context, event, selectedCards);
+
+        const gameAction = this.generatePropertiesFromContext(event.context).immediateEffect;
+        if (gameAction) {
+            const selectedArray = Array.from(selectedCards);
+            event.context.targets = selectedArray;
+            gameAction.setDefaultTargetFn(() => selectedArray);
+            context.game.queueSimpleStep(() => {
+                if (gameAction.hasLegalTarget(context)) {
+                    gameAction.resolve(null, context);
+                }
+            }, 'resolve deck search');
```

Comment:
small edit for clarity: "resolve effect on searched cards", since search has already resolved if I'm understanding right